### PR TITLE
Add performance example, introspection lib and fix timer and more

### LINKED ISSUE #17: Add performance example, introspection lib and fix timer and more
**New features:**
- performance example "iceperf"
- rename `chunkInfo `to `chunkHeader`
- remove internal port prefix and introduce port scope
- introduce introspection library
  - add size/capacity get methods for delivery FiFo
  - add port data in receiver info

**Fixes:**
- race in POSIX timer fixed

### PR DESCRIPTION
**Pull request by marthtz**
Fixes #17 

Signed-off-by: Hintz Martin (CC-AD/ESW1) <martin.hintz@de.bosch.com>

### CODE REVIEW

File: iceoryx_utils/include/iceoryx_utils/posix_wrapper/timer.hpp [hunk 1]
```diff
@@ -42,12 +47,14 @@ enum class TimerError
...
 
+
 /// @brief Interface for timers on POSIX operating systems
 /// @note Can't be copied or moved as operating system has a pointer to this object. It needs to be ensured that this
 /// object lives longer than timeToWait, otherwise the operating system will unregister the timer
+/// @concurrent not thread safe
 ///
 /// @code
-///     posix::Timer TiborTheTimer(100_ms, [&]() { fooBar++; });
+///     posix::Timer& TiborTheTimer = posix::Timer::getInstance(100_ms, [&]() { fooBar++; });
```
Comment (mossmaurice) [inline iceoryx_utils/include/iceoryx_utils/posix_wrapper/timer.hpp]: `posix::Timer::getInstance()` is not available. Could you update the example?

File: iceoryx_utils/include/iceoryx_utils/posix_wrapper/timer.hpp [hunk 2]
```diff
@@ -60,35 +67,151 @@ using namespace iox::units::duration_literals;
...
     /// @brief Creates a timer without an operating system callback
     ///
-    /// Creates a light-weight timer object that can be used with 
+    /// Creates a light-weight timer object that can be used with
     ///               * hasExpiredComparedToCreationTime()
     ///               * resetCreationTime()
     ///
     /// @param[in] timeToWait - How long should be waited?
     /// @note Does not set up an operating system timer, but uses CLOCK_REALTIME instead
+    /// @todo refactor this cTor and its functionality to a class called StopWatch
```
Comment (mossmaurice) [inline iceoryx_utils/include/iceoryx_utils/posix_wrapper/timer.hpp]: Great idea!

File: iceoryx_utils/include/iceoryx_utils/posix_wrapper/timer.hpp (hunk 1 above)
Comment (marthtz) [reply iceoryx_utils/include/iceoryx_utils/posix_wrapper/timer.hpp]: Done!

File: iceoryx_posh/source/capro/service_description.cpp [hunk 3]
```diff
@@ -70,13 +70,13 @@ ServiceDescription::ServiceDescription(const cxx::Serialization& f_serial) noexc
                      m_classHash[3],
                      m_hasServiceOnlyDescription,
                      scope);
-    if (scope > static_cast<std::underlying_type<ServiceDescription::Scope>::type>(Scope::INVALILD))
+    if (scope > static_cast<std::underlying_type<Scope>::type>(Scope::INVALID))
     {
-        m_scope = Scope::INVALILD;
```
Comment (michael-poehnl) [inline iceoryx_posh/source/capro/service_description.cpp]: @elfenpiff . So this is no poehnl and no bob or? ;-)

File: iceoryx_posh/include/iceoryx_posh/roudi/introspection_types.hpp [hunk 4]
```diff
@@ -108,6 +108,24 @@ struct PortThroughputIntrospectionFieldTopic
     cxx::vector<PortThroughputData, MAX_PORT_NUMBER> m_throughputList;
 };
 
+const capro::ServiceDescription
+    CarmaIntrospectionReceiverPortChangingDataService("CARMA", "Introspection", "ReceiverPortsData");
```
Comment (michael-poehnl) [inline iceoryx_posh/include/iceoryx_posh/roudi/introspection_types.hpp]: Should be iceoryx, or?

Comment (marthtz) [reply iceoryx_posh/include/iceoryx_posh/roudi/introspection_types.hpp]: Fixed.

File: iceoryx_examples/icedelivery/ice_publisher_simple.cpp [hunk 5]
```diff
@@ -48,14 +48,14 @@ void sending()
...
-        std::cout<< "Sending: " << ct << std::endl;
+        std::cout << "Sending: " << ct << std::endl;
 
         // pass the ownership to the middleware for sending the sample
         myTypedPublisher.publish(std::move(sample));
 
         ct++;
 
-        // Sleep some time
+        // Give the subscriber some time to process the data
```
[the remark this answers is not in the document]
Comment (marthtz) [reply iceoryx_examples/icedelivery/ice_publisher_simple.cpp]: Fixed.

File: iceoryx_examples/icedelivery/ice_subscriber_bare_metal.cpp [hunk 6]
```diff
@@ -65,7 +65,7 @@ void receiving()
             std::cout << "Not subscribed" << std::endl;
         }
 
-        // sleep
+        // Give the publisher some time to send something
```
[the remark this answers is not in the document]
Comment (marthtz) [reply iceoryx_examples/icedelivery/ice_subscriber_bare_metal.cpp]: Fixed.

File: iceoryx_examples/icedelivery/a_typed_api.hpp [hunk 7]
```diff
@@ -115,9 +117,10 @@ class TypedSubscriber
...
     {
-        const void * chunk = nullptr;
+        const void* chunk = nullptr;
 
-        // get all the chunks the FiFo holds. Maybe there are several ones if the publisher produces faster than the subscriber can process
+        // get all the chunks the FiFo holds. Maybe there are several ones if the publisher produces faster than the
+        // subscriber can process
         while (m_subscriber.getChunk(&chunk))
         {
             auto sample = static_cast<const CounterTopic*>(chunk);
```
[the remark this answers is not in the document]
Comment (marthtz) [reply iceoryx_examples/icedelivery/a_typed_api.hpp]: Fixed.

**Pull request merged**